ballerina/tests/01_client_tests.bal

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+const string username = "admin";
+const string password = "admin";
+
+const string keystorePath = "./tests/resources/docker/certs/mongodb-client.jks";
+
+@test:Config {
+    groups: ["client", "negative"]
+}
+public function testInvaliClientConfig() {
+    Client|Error mongoClient = new (invalidConfig);
+    test:assertTrue(mongoClient is ApplicationError, "Error expected when url is invalid.");
+    ApplicationError err = <ApplicationError>mongoClient;
+    test:assertEquals(err.message(), "Error occurred while initializing the MongoDB client.");
+    ApplicationError cause = <ApplicationError>err.cause();
+    test:assertEquals(cause.message(),
+        "The connection string is invalid. Connection strings must start with either 'mongodb://' or 'mongodb+srv://");
+}
+
+@test:Config {
+    groups: ["client"]
+}
+isolated function testClientInitWithNamedParameters() returns error? {
+    Client mongoClient = check new (connection = {
+        serverAddress: {
+            host: "localhost",
+            port: 27016
```

Comment:
These have default values, so we do not need to provide them here right ?

Reply:
This overrides the default port 27017.